### PR TITLE
Enable logpoint communication between debugger and agent.

### PR DESCRIPTION
**Pull request by quoctruong**
Enables logpoint on both the debugger andthe agent.

Please note that the agent is not actually writing to a log in this PR. That will come in the next PR.

### CODE REVIEW

File: src/google_cloud_debugger/google_cloud_debugger_lib/dbg_breakpoint.cc [hunk 1]
```diff
@@ -47,18 +47,21 @@ std::int32_t DbgBreakpoint::current_max_collection_size_ =
...
                                uint32_t line, uint32_t column,
+                               const bool &log_point,
                                const std::string &condition,
                                const std::vector<std::string> &expressions) {
   file_name_ = file_name;
-  id_ = id;
   std::transform(
       file_name_.begin(), file_name_.end(), file_name_.begin(),
       [](unsigned char c) -> unsigned char { return std::tolower(c); });
+  std::string underlying_file_name;
```
Comment (iantalarico) [inline src/google_cloud_debugger/google_cloud_debugger_lib/dbg_breakpoint.cc]: is this used?

File: src/google_cloud_debugger/google_cloud_debugger_lib/dbg_breakpoint.h [hunk 2]
```diff
@@ -122,6 +123,9 @@ class DbgBreakpoint : public StringStreamWrapper {
   // Sets whether this breakpoint should kill the server.
   void SetKillServer(bool kill_server) { kill_server_ = kill_server; }
 
+  // Returns whether this breakpoint is a logpoint.
+  bool GetLogPoint() const { return log_point_; }
```
Comment (iantalarico) [inline src/google_cloud_debugger/google_cloud_debugger_lib/dbg_breakpoint.h]: Maybe IsLogPoint? It's not really getting a log point

File: src/Google.Cloud.Diagnostics.Debug/Google.Cloud.Diagnostics.Debug/BreakpointExtensions.cs [hunk 3]
```diff
@@ -73,7 +74,15 @@ public static StackdriverBreakpoint Convert(this Breakpoint breakpoint)
                 EvaluatedExpressions =
                 {
                     breakpoint.EvaluatedExpressions?.Select(variable => variable.Convert()).ToList()
-                }
+                },
+
+                Expressions =
+                {
+                    breakpoint.EvaluatedExpressions?.Select(variable => variable.Name).ToList()
```
Comment (iantalarico) [inline src/Google.Cloud.Diagnostics.Debug/Google.Cloud.Diagnostics.Debug/BreakpointExtensions.cs]: Why do we need this here? This seems out of place.  Is this something we missed in the past?  If so should we have a test somewhere for this?

**Comment by quoctruong** [comment]
@iantalarico PTAL.

**Comment by quoctruong** [comment]
Thanks!

**Pull request merged**